libs/application-generic/src/services/in-memory-provider/providers/redis-provider.ts

```diff
@@ -99,7 +100,11 @@ export const getRedisInstance = (): Redis | undefined => {
   };
 
   if (port && host) {
-    return new Redis(port, host, options);
+    const redisInstance = new Redis(port, host, options);
+    if (typeof newrelic !== 'undefined' && newrelic.instrumentDatastore) {
```

Comment:
I'd also check if new relic env vars are set to indicate newrelic's enablement. Unless the newrelic object holds this information on it.

Reply:
I don't see a direct option. What about below change?
``` typescript
if (port && host) {
    const redisInstance = new Redis(port, host, options);
    const isNewRelicEnabled = typeof newrelic !== 'undefined' && newrelic.instrumentDatastore;
    const isNewRelicEnvSet = process.env.NEW_RELIC_LICENSE_KEY && process.env.NEW_RELIC_APP_NAME;

    if (isNewRelicEnabled && isNewRelicEnvSet) {
        newrelic.instrumentDatastore('Redis', () => redisInstance);
    }
}
```

Reply:
Perfect